telemetry_device_main/decryptor/IcdPacketDecryptor.cs

```diff
@@ -65,9 +65,17 @@ private void GenerateParameters(List<IcdType> icdRows, ref Dictionary<string, (i
             int corValue = -1;
             foreach (IcdType row in icdRows)
             {
+                if (row.GetLocation() == -1)
+                    continue;
+                if (row.GetCorrValue() != -1 && corValue != row.GetCorrValue())
+                    continue;
+
                 byte[] rowValue = GetAccurateValue(row, packet);
                 CreateMask(row.GetMask(), ref rowValue[0]);
```

Comment:
why rowValue in location 0?

Reply:
assuming masking is preformed on 8 bit values so if there is mask it will be the first and only byte in current row